pkg/vsphere/guestinfo/encode.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guestinfo
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	// DefaultTagName value
+	DefaultTagName = "vic"
+	// DefaultPrefix value
+	DefaultPrefix = ""
+)
+
+const (
+	// Invalid value
+	Invalid = 1 << iota
+	// Hidden value
+	Hidden
+	// ReadOnly value
+	ReadOnly
+	// ReadWrite value
+	ReadWrite
+	// NonPersistent value
+	NonPersistent
+	// Volatile value
+	Volatile
```

Comment:
You were talking about exporting volatile objects via getter/setters, we can do that if we still want that. Or we can drop and add it back when we need them